frontend/packages/kubevirt-plugin/src/utils/grammar.ts

```diff
@@ -0,0 +1,20 @@
+import * as _ from 'lodash';
+
+export const joinGrammaticallyListOfItems = (items: string[]) => {
+  const result = items.join(', ');
+  const lastCommaIdx = result.lastIndexOf(',');
+
+  return items.length > 1 && lastCommaIdx >= 0
+    ? `${result.substr(0, lastCommaIdx)} and${result.substr(lastCommaIdx + 1)}`
+    : result;
+};
+
+export const makeSentence = (sentence: string, capitalize = true) => {
+  const result = capitalize ? _.capitalize(sentence) : sentence;
+  return !result || result.charAt(result.length) === '.' ? result : `${result}.`;
+};
+
+export const addMissingSubject = (sentence: string, subject: string) => {
+  const c = sentence ? sentence.charAt(0) : '';
+  return c.toLowerCase() !== c ? sentence : `${_.capitalize(subject)} ${sentence}`;
```

Comment:
Looks like copy&paste from https://github.com/kubevirt/web-ui-components/pull/524 .

In any case, I don't like this approach, looks to me like unnecessary complication for future potential internationalization.

Reply:
> Looks like copy&paste from [kubevirt/web-ui-components#524](https://github.com/kubevirt/web-ui-components/pull/524) .

it is because it is used in two places and we are moving it to this repo

> 
> In any case, I don't like this approach, looks to me like unnecessary complication for future potential internationalization.

This gives better user experience and can be further customized once i18n is implemented.

Reply:
>> it is because it is used in two places and we are moving it to this repo

That's the same story for other functionality. If we start copy&paste such code, we will need to maintain it twice. If particular code needs to live in web-ui-components and so it is not preferred to pass it from openshift/console, it would be better to `import` it.

Once we get to the point of removing `web-ui-components` from dependency list, we will move these bits all together.

Speaking of this particular bit, it is being newly introduced.

Reply:
> This gives better user experience and can be further customized once i18n is implemented.

I will not block merging on this but I consider this approach as not a good practice.I believe same user experience can be achieved with different implementation.

Reply:
> > > it is because it is used in two places and we are moving it to this repo
> 
> That's the same story for other functionality. If we start copy&paste such code, we will need to maintain it twice. If particular code needs to live in web-ui-components and so it is not preferred to pass it from openshift/console, it would be better to `import` it.
> 
> Once we get to the point of removing `web-ui-components` from dependency list, we will move these bits all together.
> 
> Speaking of this particular bit, it is being newly introduced.

We are already moving portions of web-ui-components here